evalml/pipelines/components/transformers/preprocessing/datetime_featurizer.py

```diff
@@ -7,19 +7,33 @@
 
```

Comment:
This implementation takes 150 ms to create features for 150k rows. I think that's reasonable.

![image](https://user-images.githubusercontent.com/41651716/100675023-2bbc2a80-3334-11eb-9000-be3a8cd655eb.png)